docs/distribution/README.md

```diff
@@ -0,0 +1,923 @@
+1. Link **acme-rockets** signature to the `net-monitor:v1` container image
+
+```HTTP
+PUT https://localhost:6000/v2/net-monitor/manifests/sha256:2235d2d22ae5ef400769fa51c84717264cd1520ac8d93dc071374c1be49cc77c/signatures/sha256:007170c33ebc4a74a0a554c86ac2b28ddf3454a5ad9cf90ea8cea9f9e75a153b
+```
+
+**Pros with this approach:**
+
+* Using a unique API enables RBAC to the API, without having to parse the content to determine the role check.
+
+**Cons with this approach:**
+
+* A new linking api, unique to signatures
+* The client must make two calls to achieve a single operation of uploading a signature object, which by definition has the linking information
+* If the signature linking API fails, additional garbage collection of a signature manifest must be cleaned up
+
+### Linking Signatures - Option 3: Signature Upload API
+
+In this option the signature artifact (manifest or index) is uploaded through a new signature API.
+
+**Pros with this approach:**
+
+* The signature artifact upload and role check are coupled to a signature API
+
+**Cons with this approach:**
+
+* Signature upload, which is just another artifact type, is uploaded differently than other artifacts.
+
+### Linking Signatures - Option 4: Utilize OCI Index PUT
+
+Utilizing the OCI Index, the manifest list is used to track dependencies.
+
+**Pros with this approach:**
+
+* Existing Index parsing logic is used to track despondencies
+* No additional distribution-spec APIs are required
+
+**Cons with this approach:**
+
+* OCI Index would need to support a config object:  [Add Index Support for Artifact Type #806](https://github.com/opencontainers/image-spec/issues/)
+
+## Signature Discovery
+
+Once a signature artifact is in a registry and linked to its target artifact, how is it retrieved?
+
+The following options are offered:
+
+1. [Option 1: Signature Listing API](#signature-discovery---option-1-rest-api-standard-paging)
+1. [Option 2: Generic Reference Listing API](#signature-discovery---option-2-generic-reference-listing-api)
+
+### Signature Discovery - Option 1: Signature Listing API
+
+Similar to the [_tags api][tags-api], a new signatures API is proposed. The signatures API uses the digest as the path to find all signatures related to said digest.
+In the below example, the `net-monitor:v1` tag has a digest of: `sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042`
+
+```HTTP
+GET /v2/<name>/manifests/sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042/signatures/
+```
+
+The response will be in the following format:
+
+```HTTP
+200 OK
+Content-Type: application/json
+{
+  "digest": "sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042",
+  "signatures": [
+    {
+      "digest": "sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042",
+      "mediaType": "application/vnd.oci.image.index.v1+json",
+      "size": "1024",
+      "signature-typ": "x509"
```

Comment:
Is it necessary to include `signature-typ` in the response?

If it's removed, the item in `signatures` list is more consistent with definition in:

https://github.com/opencontainers/image-spec/blob/master/descriptor.md

Reply:
Yeah, this is the problem noted above at: https://github.com/notaryproject/nv2/pull/10#discussion_r481469759
we need more focus here on whether we'll have a generic api, or a signature specific api.

Reply:
I've captured it here:: #15